docs/evm/cadence/batched-evm-transactions.md

```diff
@@ -0,0 +1,641 @@
+applications. So let's get started!
+
+## Components
+
+As mentioned in the [Overview](#overview), this guide involves three main actions:
+
+- Wrapping FLOW as WFLOW
+- Approving an ERC721 to transfer WFLOW in exchange for an NFT mint
+- Minting an ERC721 token
+
+Before interacting with these contracts, let's dig in a bit more into the components of this guide.
+
+### Wrap FLOW as WFLOW
+
+On Flow EVM, FLOW is the native currency and similar to other EVM platforms, the native currency is not accessible as an
+ERC20 token. To interact with ERC20 contracts, you need to wrap FLOW as WFLOW (Wrapped FLOW). This is Flow's equivalent
+of WETH on Ethereum.
+
+:::tip
+
+You can find WFLOW deployed to `0xd3bF53DAC106A0290B0483EcBC89d40FcC961f3e` on Flow
+[Testnet](https://evm-testnet.flowscan.io/token/0xd3bF53DAC106A0290B0483EcBC89d40FcC961f3e?tab=contract) &
+[Mainnet](https://evm.flowscan.io/token/0xd3bF53DAC106A0290B0483EcBC89d40FcC961f3e?tab=contract) and source code in the
+[`@onflow/flow-sol-utils` repository](https://github.com/onflow/flow-sol-utils).
+
+:::
+
+### Approve ERC721 Transfer
+
+Our example `MaybeMintERC721` contract accepts WFLOW in exchange for minting an NFT. However, the contract cannot move
+WFLOW without your permission. To allow the contract to move your WFLOW, you must approve the contract to transfer
+enough of your WFLOW to mint the NFT.
+
+### Mint ERC721 Token
+
+Finally, we'll mint an ERC721 token using the `MaybeMintERC721` contract. This contract has a 50% chance of failing,
+simulating a real-world scenario where purchasing an NFT might fail - say a listing was purchased before your
+transaction was processed.
+
+Importantly, if this transaction fails, we want to revert the entire sequence of transactions. After all, you wrapped
+FLOW to WFLOW and approved the ERC721 transfer specifically to mint this NFT. If the mint fails, you want to unwind
+everything. This is where batching EVM transactions using Cadence is extremely powerful.
+
+## Interacting with the Contracts
+
+Before taking the easy route, let's first interact with the contracts individually to better understand the process and
+status quo user experience. Realistically, this is your only option for completing the whole process on other EVM
+platforms.
+
+:::tip
+
+Recall in [Prerequisites](#prerequisites) that you need to have both MetaMask and Flow Wallet installed and funded with
+Testnet FLOW. Make sure you've done so before proceeding.
+
+:::
+
+### Using MetaMask
+
+#### 1. Wrap FLOW
+
+Our first action will be to wrap enough FLOW to cover the cost of minting the `MaybeMintERC721` token. To do this, we'll
+interact with the `WFLOW` contract on Testnet. There are a number of ways we could interact with this contract - Remix
+IDE, Foundry's CLI, Hardhat, etc. - but for the purposes of this guide, we'll use the [Flowscan EVM block
+explorer](https://www.evm-testnet.flowscan.io/).
+
```

Comment:
I'd add a :::warning here about the evm vs. Cadence versions of FlowScan

Reply:
Fixed in [c19893b](https://github.com/onflow/docs/pull/983/commits/c19893b7ade1cd95602c6ab00f9c81bd77b149b4)